src/test/docker/docker-run-checks.sh

```diff
@@ -128,6 +128,45 @@ if [[ "$INSTALL_ONLY" == "t" ]]; then
                make clean && 
                make -j${JOBS}" \
 	|| (docker rm tmp.$$; die "docker run of 'make install' failed")
+
+elif [[ -n "$S3_ACCESS_KEY_ID" ]]; then
```

Comment:
If the only difference here is to add the `S3_*` environment vars to the `docker run` commandline, then we probably don't need a whole separate invocation under `elif [[ -n "$S3_ACCESS_KEY_ID" ]]`. Just include those vars on the existing invocation.